rskj-core/src/test/java/co/rsk/peg/bitcoin/UtxoUtilsTest.java

```diff
@@ -0,0 +1,231 @@
+package co.rsk.peg.bitcoin;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import co.rsk.bitcoinj.core.Coin;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.spongycastle.util.encoders.Hex;
+
+class UtxoUtilsTest {
+
+    private final static Coin MAX_BTC = Coin.valueOf(21_000_000, 0);
+
+    private static Stream<Arguments> validOutpointValues() {
```

Comment:
This is extremely hard to read and follow. Think you can find a way to simplify?

Reply:
Extracting each case into separate method.

Reply:
agree with marcos